src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala

```diff
@@ -198,6 +198,11 @@ private[internal] trait TypeConstraints {
         val bound: Type = if (up) tparam.info.bounds.hi else tparam.info.bounds.lo
         //Console.println("solveOne0(tv, tp, v, b)="+(tvar, tparam, variance, bound))
         var cyclic = bound contains tparam
+        // This is attempting to reorder the solving of TypeVars such that tvs which are
+        // mentioned in the active bound are solved first.
+        // 1. Is this reordering desirable? Should solving always be left to right instead?
+        // 2. Should this take account of the variance of tparam2 and flip the bounds if
+        //    necessary?
```

Comment:
When would it be necessary to flip the bounds?

Reply:
Flip the bounds depending on whether the param is co/inv vs. contra variant.

Reply:
`up` depends on the variance of `tparam` and the variance of `tparam2` shouldn't matter I think.

Reply:
I'd like to be convinced of that :-)